src/Microsoft.DotNet.Build.VSTSBuildsApi/VSTSBuildClient.cs

```diff
@@ -0,0 +1,234 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace VstsBuildsApi
+{
+    public class VstsBuildClient
+    {
+        private string _collectionIdentifier;
+        private string _credentials;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="collectionIdentifier">collection identifer for build definitions</param>
+        /// <param name="credentials">credentials used for basic rest api authentication</param>
+        public VstsBuildClient(string collectionIdentifier, string credentials)
+        {
+            if(string.IsNullOrWhiteSpace(collectionIdentifier))
+            {
+                throw new Exception("Required parameter (collectionIdentifier) cannot be null or white space.");
+            }
+            if(!Uri.IsWellFormedUriString(collectionIdentifier, UriKind.Relative))
+            {
+                throw new Exception(string.Format("collectionIdentifier '{0}' contains invalid characters.  collectionIdentifier must contain only valid uri characters.", collectionIdentifier));
+            }
+
+            _collectionIdentifier = collectionIdentifier;
+            _credentials = credentials;
+        }
+
+        /// <summary>
+        /// Load a local build definition, combine the uniqueIdentifier with the definition name, 
+        /// check VSTS for a matching definition name, if present, update that definition with the
+        /// local build definition, otherwise create a new definition.
+        /// </summary>
+        /// <param name="stream">Stream to a VSTS build definition</param>
+        /// <returns>Created or updated build Id</returns>
+        public async Task<string> CreateOrUpdateDefinitionAsync(Stream stream)
+        {
+            JObject definition = GetDefinition(stream);
+            return await CreateOrUpdateDefinitionAsync(definition);
```

Comment:
Consider using .ConfigureAwait(false)
This is a recommended practice for libraries that expose async methods and don't need to flow context.